src/github.com/getlantern/balancer/strategy.go

```diff
@@ -97,9 +100,11 @@ func (s *dialerHeap) Pop() interface{} {
 	return x
 }
 
-func (s *dialerHeap) AuthTokens() (tokens []string) {
+func (s *dialerHeap) onRequest(req *http.Request) (tokens []string) {
 	for _, d := range s.dialers {
-		tokens = append(tokens, d.AuthToken)
+		if d.OnRequest != nil {
+			d.OnRequest(req)
+		}
```

Comment:
Looks like we can just scrap the return value...surprised go let's that compile actually...